10 topic page

MFStevenson opened the pull request.

MFStevenson committed:
post comment
post works and re-renders after submit
not implemented optimistically currently as unsure how to give unque id as this happens during post

MFStevenson committed:
allows user to delete a comment

MFStevenson committed:
allows user to view topics
user can view a list of topics and go to relevant articles

popatre commented on src/pages/CommentsPage.jsx:
```diff
@@ -1,27 +1,44 @@
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const postBody = { username: "jessjelly", comment: input };
+    postComment(article_id, postBody)
```
Did you branch from a branch?

popatre commented on src/pages/CommentsPage.jsx:
```diff
@@ -34,16 +51,33 @@ const CommentsPage = () => {
+        <form onSubmit={handleSubmit}>
+          <label>
+            Comment{" "}
+            <input
```
I think you've branched from a branch - im not sure why this is in this commit. Think you're going to have some tricky merge conflicts

popatre commented on src/utils/api.js:
```diff
@@ -24,6 +24,10 @@ export const patchArticleVotes = (article_id, votes) => {
 
-export const postComment = (article_id) => {
-  return ncNewsApi.post(`/api/articles/${article_id}/comments`);
+export const postComment = (article_id, postBody) => {
```
Its really difficult to check this ticket as it seems to have loads of stuff to do with other tickets as well

popatre reviewed (changes requested):
This PR was really hard to look at as it seems to have lots of stuff committed from other open PRs. 

Topic PR wise - how can i send someone all the articles to do with football? At the moment, they have to go to one page, then click a button. 
For example, heres all the business news articles from the BBC - https://www.bbc.co.uk/news/business

You can go straight there and i dont need to tell you to go one place, then find a button to click. 

 Because of the current set up, they disappear when you refresh the page as well which isnt great UI/UX

MFStevenson committed:
actions feedback
changes navbar to use links instead of buttons
adds routing to articles so that topic filters are on different url

nc-modou reviewed (approved):
Good response to Jonathan's comments! Topics now has it's own page which works great, and the change to the url means you can now access them by URL directly. Feel free to merge!
